src/router/layer.ts

```diff
@@ -46,7 +47,7 @@ Layer.prototype.handle_error = async function handle_error(
   }
 
   try {
-    await fn(error, req, res, next);
+    await fn(error, requestProxy(req), res, next);
```

Comment:
Would this not be better placed earlier on, such as in `app.handle()`, rather than wrapping the request in a proxy for _every_ layer in the middleware stack. Concerned we'd be constructing new proxies every layer and this will have a bad impact on performance?